### PR TITLE
authentification

### PR DESCRIPTION
**Pull request by vic778**
- setup the app
- I generate models
- create models
- Setup dvise
- Generate views for devise
- Generaate user model
- Generate moigration roel to user
- Generate the homes page and make it as the root_path
- Permit username to devise
- Generate food controller
- Desable turbo for devise
- Desable turbo for devise
- Desable turbo for devise
- Desable turbo for devise
- Desable turbo for devise

### CODE REVIEW

**Comment by Alfredbis29** [comment]
Hello @vic778  , I tried to establish users' roles using Pundit, but it didn't work. Ideally, the user login should only grant access to his relevant features depending on whether he is an admin or not. can you please give me some assistance on this ?

**Comment by vic778** [comment]
Hey @Alfredbis29  How are you doing? can you fix this error, please? I would like to fix the bug you are facing but once pull the PR I got this error when running the server which blocks me to proceed. 

Screenshot from 2022-11-17 11-15-33

File: db/migrate/20221118102619_add_admin_column_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddAdminColumnToUser < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :admin, :boolean, default: "false"
+  end
+end
```
Comment (vic778) [inline db/migrate/20221118102619_add_admin_column_to_user.rb:5]: Kindly we have already the role for the user. this migration trow a conflict

File: app/views/foods/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<div class="mb-3">
+  <label for="exampleFormControlInput1" class="form-label">Food Image</label>
+  <input type="food_price" class="form-control" id="exampleFormControlInput1" placeholder="Image_url">
+</div>
+<div class="mb-3">
+  <label for="exampleFormControlTextarea1" class="form-label">Description</label>
+  <textarea class="form-control" id="exampleFormControlTextarea1" rows="3"></textarea>
+</div>
+ <button type="submit" class="btn btn-primary">Order</button>
+</form>
```
Comment (vic778) [inline app/views/foods/new.html.erb]: Kindly the form is not creating the food.

File: config/database.yml [hunk 3]
```diff
@@ -17,17 +17,17 @@
...
 development:
   <<: *default
   database: Food_delivery_app_development
-  # username: postgres
-  # # password: alfred
+  username: postgres
+  password: alfred
   # The specified database role being used to connect to postgres.
   # To create additional roles in postgres see `$ createuser --help`.
   # When left blank, postgres will use the default role. This is
```
Comment (vic778) [inline config/database.yml]: TO avoid conflicts, kindly ignore the database.yml

File: .gitignore [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 # If you find yourself ignoring temporary files generated by your text editor
 # or operating system, you probably want to add a global ignore instead:
 #   git config --global core.excludesfile '~/.gitignore_global'
 
 # Ignore bundler config.
 /.bundle
-
+/.Gemfile
+/.ruby-version
+/.database.yml
```
Comment (vic778) [inline .gitignore:10]: Those files can not be ignored like that because they are not starting well.

**Review by vic778 — COMMENTED** [review]
Hey @Alfredbis29  Have a look at some requests

File: app/views/foods/new.html.erb [hunk 5]
```diff
@@ -1,19 +1,9 @@
...
\ No newline at end of file
+<%= form_for :person do |f| %>
+  Food name: <%= f.text_field :food_name %><br />
+  Price : <%= f.text_field :price %><br />
+  Description : <%= f.text_area :description %><br />
+  Image : <%= f.fields_for :exhibition_images do |f| %>
+    <%= f.file_field :image, help: "Ensure images are minimum 400x400px"  %><br />
+  Admin?    : <%= f.check_box :admin %><br />
+  <%= f.submit %>
+<% end %>
```
Comment (vic778) [inline app/views/foods/new.html.erb]: Weldon so far. kindly tag the model instead of the column.

**Review by vic778 — COMMENTED** [review]
Hey @Alfredbis29  well done so far. Kindly read the feedback

**Pull request merged**